ui/admin/tests/acceptance/roles/create-test.js

```diff
@@ -61,75 +63,86 @@ module('Acceptance | roles | create', function (hooks) {
     urls.roles = `/scopes/${instances.scopes.org.id}/roles`;
     urls.role = `${urls.roles}/${instances.role.id}`;
     urls.newRole = `${urls.roles}/new`;
-    urls.orgScope = `/scopes/${instances.scopes.org.id}`;
+    getRolesCount = () => this.server.schema.roles.all().models.length;
   });
 
   test('can create new role', async function (assert) {
-    const rolesCount = this.server.db.roles.length;
+    const rolesCount = getRolesCount();
     await visit(urls.newRole);
-    await fillIn('[name="name"]', 'role name');
-    await click('[type="submit"]');
-    assert.strictEqual(this.server.db.roles.length, rolesCount + 1);
+
+    await fillIn(commonSelectors.FIELD_NAME, commonSelectors.FIELD_NAME_VALUE);
+    await click(commonSelectors.SAVE_BTN);
+
+    assert.strictEqual(getRolesCount(), rolesCount + 1);
   });
 
   test('Users can navigate to new roles route with proper authorization', async function (assert) {
     await visit(urls.roles);
+
     assert.ok(
       instances.scopes.org.authorized_collection_actions.roles.includes(
         'create',
       ),
     );
-    assert.ok(find(`[href="${urls.newRole}"]`));
+    assert.dom(selectors.NEW_ROLE_BTN).isVisible();
+    assert.dom(commonSelectors.IAM_SIDEBAR_NAV_LINK(urls.roles)).isVisible();
+    assert.dom(commonSelectors.HREF(urls.newRole)).isVisible();
```

Comment:
I added extra checks here because we really wanted to verify multiple things with this test.

1. That the "New Role" button was visible
2. The url linked to that button is correct
3. The url for navigating to Roles is also visible

I also did the inverse in the test below in the case of missing the correct `authorized_collection_actions`

Reply:
Nice! leaving better than we find it! 💪